controller.go

```diff
@@ -0,0 +1,208 @@
+	default:
+		return CommandResponse{ErrorResponse(fmt.Errorf("unsupported command"))}
+	}
+}
+
+// bind triggers the bind process which is triggerred by a client
+// Only pre-defined owner DID is allowed to initiate the binding process.
+func (c *Controller) bind(did string) []byte {
+	if did != c.ownerDID {
+		return ErrorResponse(errors.New("illegal owner"))
+	}
+
+	bound, err := c.IsPodBound()
+	if err != nil {
+		return ErrorResponse(err)
+	}
+
+	if bound {
+		log.Warn("node has bound")
+		return ErrorResponse(fmt.Errorf("node has bound"))
+	}
+
+	b, err := utils.GenerateRandomBytes(4)
+	if err != nil {
+		return ErrorResponse(err)
+	}
+
+	nonce := hex.EncodeToString(b)
+	nowString := fmt.Sprint(int64(time.Now().UnixNano()) / int64(time.Millisecond))
+	signature, err := key.Sign(c.Identity.PrivateKey, nonce+nowString)
+	if err != nil {
+		return ErrorResponse(err)
+	}
+
+	c.bindingNonce = nonce
+
+	return ObjectResponse(map[string]string{
+		"identity":  c.Identity.DID,
+		"nonce":     nonce,
+		"timestamp": nowString,
+		"signature": signature,
+	})
+}
+
+// bindACK process the client response of a binding process. It checks the nonce
+// and the signature using owner DID.
+func (c *Controller) bindACK(did string, ackParams BindACKParams) []byte {
+	defer func() {
+		c.bindingNonce = ""
+	}()
+
+	if did != c.ownerDID {
+		return ErrorResponse(errors.New("illegal owner"))
+	}
+
+	if c.bindingNonce == "" {
+		err := fmt.Errorf("binding request not found")
+		log.WithError(err).Error("fail to bind account")
+		return ErrorResponse(err)
+	}
+
+	if !key.VerifySignature(did, c.bindingNonce+ackParams.Timestamp, ackParams.Signature) {
+		err := fmt.Errorf("invalid binding ack signature")
+		log.WithError(err).Error("fail to bind account")
+		return ErrorResponse(err)
+	}
+
+	if err := c.BindAccount(); err != nil {
+		log.WithError(err).Error("fail to bind account")
+		return ErrorResponse(err)
+	}
+	log.WithField("did", did).Println("bind account successfully")
+
+	return ObjectResponse(map[string]string{
+		"status": "ok",
+	})
+}
+
+// bitcoinRPC runs bitcoind rpc for clients
+func (c *Controller) bitcoinRPC(did string, bitcoindParams BitcoindRPCParams) []byte {
+	bound, err := c.IsPodBound()
+	if err != nil {
+		return ErrorResponse(err)
+	}
+
+	if !bound {
+		return ErrorResponse(fmt.Errorf("node is not bound"))
+	}
+
+	var reqBody bytes.Buffer
+
+	if err := json.NewEncoder(&reqBody).Encode(bitcoindParams); err != nil {
+		return ErrorResponse(err)
+	}
+
+	// log.WithField("reqBody", reqBody.String()).Info("requset body")
+	req, err := http.NewRequest("POST", viper.GetString("bitcoind.rpcconnect"), &reqBody)
+	if err != nil {
+		return ErrorResponse(err)
+	}
+	req.SetBasicAuth(viper.GetString("bitcoind.rpcuser"), viper.GetString("bitcoind.rpcpassword"))
+
+	resp, err := c.httpClient.Do(req)
+	if err != nil {
+		return ErrorResponse(err)
+	}
+	defer resp.Body.Close()
+
+	var responseBody json.RawMessage
+	if resp.StatusCode != 401 {
```

Comment:
Why is 401 treated differently?

Reply:
It is because both `400`  and `500` return the error object where the `401` doesn't. There might be other non-tested cases. We will fix that once it is found.